### ISSUE TITLE
Template compile fails to notice an invalid template

### ISSUE THREAD

**Issue by ihrwein** [sunng87/handlebars-rust#62]
First of all, thanks for this library :)

I noticed that this unit test would panic as Template successfully compiles `source`:

```rust
#[test]
fn test_invalid_template_cannot_be_compiled() {
    let source = "{{invalid}".to_string();
    let _ = Template::compile(source).err().unwrap();
}
```
Is this the intended behavior?

I'm in the middle of writing unit tests for my library and I wanted to include one when the template cannot be compiled. I tried this one first but the compilation was successful, so I came here.

Update: `{invalid}}` cannot be compiled.
Also, could you provide me an example which causes a RenderError? I tried `{{log}}` (without any parameters), but it seems like it's ok.

**Comment by sunng87** [comment]
@ihrwein Thanks for reporting. This is actually an issue that unclosed expression is not reported. I will fix it asap.

**Comment by sunng87** [comment]
and `{{log}}` without parameter will raise a `RenderError` when you render something.

**Comment by sunng87** [comment]
@ihrwein this has been fixed in 0.14.0, please update and verify it.
